oxd-server/src/main/java/org/gluu/oxd/server/persistence/service/GluuPersistenceService.java

```diff
@@ -0,0 +1,306 @@
+            LOG.debug("Expired_object created successfully. Object : {} ", obj.getKey());
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to create ExpiredObject: {} ", obj.getKey(), e);
+        }
+        return false;
+    }
+
+    public boolean update(Rp rp) {
+        try {
+            RpObject rpObj = new RpObject(getDnForRp(rp.getOxdId()), rp.getOxdId(), Jackson2.serializeWithoutNulls(rp));
+            this.persistenceEntryManager.merge(rpObj);
+            LOG.debug("RP updated successfully. RP : {} ", rpObj);
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to update RP: {} ", rp, e);
+        }
+        return false;
+    }
+
+    public Rp getRp(String oxdId) {
+        try {
+            RpObject rpFromGluuPersistance = getRpObject(oxdId, new String[0]);
+
+            Rp rp = MigrationService.parseRp(rpFromGluuPersistance.getData());
+            if (rp != null) {
+                LOG.debug("Found RP id: {}, RP : {} ", oxdId, rp);
+                return rp;
+            }
+            LOG.error("Failed to fetch RP by id: {} ", oxdId);
+            return null;
+        } catch (Exception e) {
+            LOG.error("Failed to update oxdId: {} ", oxdId, e);
+        }
+        return null;
+    }
+
+    private RpObject getRpObject(String oxdId, String... returnAttributes) {
+        return (RpObject) this.persistenceEntryManager.find(getDnForRp(oxdId), RpObject.class, returnAttributes);
+    }
+
+    public ExpiredObject getExpiredObject(String key) {
+        try {
+            ExpiredObject expiredObject = getExpiredObject(key, null);
+            expiredObject.setType(ExpiredObjectType.fromValue(expiredObject.getTypeString()));
+            if (expiredObject != null) {
+                LOG.debug("Found ExpiredObject id: {} , ExpiredObject : {} ", key, expiredObject);
+                return expiredObject;
+            }
+            LOG.error("Failed to fetch ExpiredObject by id: {} ", key);
+            return null;
+        } catch (Exception e) {
+            if (((e instanceof EntryPersistenceException)) && (e.getMessage().contains("Failed to find entry"))) {
+                LOG.warn("Failed to fetch ExpiredObject by id: {}. {} ", key, e.getMessage());
+                return null;
+            }
+            LOG.error("Failed to fetch ExpiredObject by id: {} ", key, e);
+        }
+        return null;
+    }
+
+    private ExpiredObject getExpiredObject(String key, String... returnAttributes) {
+        return (ExpiredObject) this.persistenceEntryManager.find(getDnForExpiredObj(key), ExpiredObject.class, returnAttributes);
+    }
+
+    public boolean isExpiredObjectPresent(String key) {
+        return getExpiredObject(key) != null;
+    }
+
+    public boolean removeAllRps() {
+        try {
+            this.persistenceEntryManager.remove(String.format("%s,%s", new Object[]{getRpOu(), getOxdDn()}), RpObject.class, null, 100);
```

Comment:
We need configuration property `persistence-manager-remove-count: 1000`

Reply:
Done